File: fixtures/programming/java/UserService.java
package com.example.service;

import java.util.Map;
import java.util.HashMap;
import java.util.List;
import java.util.Optional;

/**
 * Service for managing users — supports CRUD operations.
 */
public class UserService {
    private final Map<Long, User> store = new HashMap<>();

    /**
     * Adds a user to the store.
     * @param user the user to add
     */
    public void addUser(User user) {
        store.put(user.getId(), user);
    }

    /** Retrieves a user by ID. */
    public Optional<User> getUser(long id) {
        return Optional.ofNullable(store.get(id));
    }

    /** Lists all users. */
    public List<User> listUsers() {
        return List.copyOf(store.values());
    }
}

File: fixtures/programming/csharp/UserService.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Example.Services;

/// <summary>
/// Manages user accounts — supports lookup and creation.
/// </summary>
public class UserService
{
    private readonly Dictionary<int, User> _store = new();

    /// <summary>Adds a user to the store.</summary>
    public void Add(User user)
    {
        _store[user.Id] = user;
    }

    /// <summary>Finds a user by ID, or null if not found.</summary>
    public User? FindById(int id)
    {
        return _store.GetValueOrDefault(id);
    }

    /// <summary>Returns all users sorted by name.</summary>
    public IReadOnlyList<User> ListAll()
    {
        return _store.Values.OrderBy(u => u.Name).ToList();
    }
}

/// <summary>Represents a user in the system.</summary>
public record User(int Id, string Name, string Email);
